tls/irc_client_tls.ml

```diff
@@ -0,0 +1,48 @@
+open Lwt.Infix
+
+module Io = struct
+  type 'a t = 'a Lwt.t
+  let (>>=) = Lwt.bind
+  let return = Lwt.return
+
+  type file_descr = {
+    ic: Tls_lwt.ic;
+    oc: Tls_lwt.oc;
+  }
+
+  type inet_addr = string
+
+  type config = Tls.Config.client
+
+  let default_config : Tls.Config.client =
+    Tls.Config.client ~authenticator:X509.Authenticator.null ()
```

Comment:
well, `irc-client` is a library, so any user can plug their own `config` argument in it. By default I can't figure out how to have better authentication since what you link requires some user input or specific certificates :-/